<a href="https://colab.research.google.com/github/adelriscom/DataScience/blob/main/Lab4_Alexander_Del_Risco.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# First, we import packages used in this lab

In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from scipy import stats
from IPython.display import display

In [12]:
df_insurance = pd.read_csv('/content/drive/MyDrive/DataScience_UWinnipeg/insurance.csv')
display(df_insurance.head(10))

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520
5   31  female  25.740         0     no  southeast   3756.62160
6   46  female  33.440         1     no  southeast   8240.58960
7   37  female  27.740         3     no  northwest   7281.50560
8   37    male  29.830         2     no  northeast   6406.41070
9   60  female  25.840         0     no  northwest  28923.13692

In [10]:
print(df_insurance.shape)

(1338, 7)


In [11]:
def check_variables(df: pd.DataFrame):
  cols = df.columns
  quantitative_cols = df._get_numeric_data().columns
  categorical_cols = list(set(cols) - set(quantitative_cols))
  return categorical_cols, list(set(quantitative_cols))

cat_cols, quan_cols = check_variables(df_insurance)
print(f'categorical columns: {cat_cols}')
print(f'quantitative columns: {quan_cols}')

categorical columns: ['region', 'sex', 'smoker']
quantitative columns: ['bmi', 'children', 'charges', 'age']


# Follow the lab introduction, investigate the following statement based on probability.

># Calculate the average insurance charges for male and female

>H(0): The mean insurance charges for men and women are equal.

>H(1): The mean insurance charges for men are higher than the mean insurance charges for women.

In [29]:
male_charge_avg = df_insurance[df_insurance.sex=='male']['charges'].mean()
female_charge_avg = df_insurance[df_insurance.sex=='female']['charges'].mean()
female_charge_std = df_insurance[df_insurance.sex=='female']['charges'].std()
z_score = (male_charge_avg - female_charge_avg)/female_charge_std

p_value = 1 - stats.norm.cdf(z_score)

print("Z-score:", z_score)
print("p-value:", p_value)

Z-score: 0.12464814938936905
p-value: 0.45040105396168617


>>The calculated z-score for the hypothesis that overall, insurance charges for men are higher than insurance charges for women is 0.1246, which indicates that the difference in means between the two groups is less than one-tenth of a standard deviation. The p-value of 0.4504 suggests that this difference is not statistically significant at a significance level of 0.05, meaning that we cannot reject the null hypothesis that there is no significant difference in insurance charges between men and women.

# An insurance expert said that overall, the insurance charges for people in ‘southwest’ region is higher than the insurance charges for people in the ‘northwest’ region


>> Null hypothesis (H0): The insurance charges for people in the ‘southwest’ region are the same or lower than the insurance charges for people in the ‘northwest’ region.

>>Alternative hypothesis (H1): The insurance charges for people in the ‘southwest’ region are higher than the insurance charges for people in the ‘northwest’ region.

In [31]:
sw_charges_avg = df_insurance[df_insurance.region == 'southwest']['charges'].mean()
nw_charges_avg = df_insurance[df_insurance.region == 'northwest']['charges'].mean()
nw_charge_std = df_insurance[df_insurance.region=='northwest']['charges'].std()
z_score = (sw_charges_avg - nw_charges_avg)/nw_charge_std

p_value = 1 - stats.norm.cdf(z_score)

print("Z-score:", z_score)
print("p-value:", p_value)

Z-score: -0.006379717301052088
p-value: 0.5025451217036413


>>The Z-score for the hypothesis "the insurance charges for people in 'southwest' region is higher than the insurance charges for people in the 'northwest' region" is -0.0064, and the p-value is 0.5025. Since the p-value is greater than 0.05, we do not have enough evidence to reject the null hypothesis that there is no significant difference in insurance charges between the two regions. Therefore, we cannot conclude that the insurance charges for people in the 'southwest' region are higher than the insurance charges for people in the 'northwest' region.

# An insurance expert said that overall, the insurance charges for people having 0 children is higher than the insurance charges for people having children.

>>Null hypothesis 
H0: The mean insurance charges for people with no children is equal to or lower than the mean insurance charges for people with one or more children.
Alternative hypothesis 

>>H1: The mean insurance charges for people with no children is higher than the mean insurance charges for people with one or more children.

In [36]:
no_children_charges_avg = df_insurance[df_insurance.children == 0]['charges'].mean()
children_charges_avg = df_insurance[df_insurance.children > 0]['charges'].mean()
children_charge_std = df_insurance[df_insurance.children > 0]['charges'].std()
z_score = (no_children_charges_avg - children_charges_avg) / children_charge_std

p_value = 1 - stats.norm.cdf(z_score)

print("Z-score:", z_score)
print("p-value:", p_value)

Z-score: -0.13049312676053818
p-value: 0.551911854151137


>>Based on the results of the hypothesis test, we cannot reject the null hypothesis at a significance level of 0.05. Therefore, we cannot conclude that the insurance charges for people having 0 children is higher than the insurance charges for people having children. The p-value of 0.5519 is greater than the significance level of 0.05, indicating that the difference between the two groups may have occurred by chance.